### PR TITLE
[dev] AB#36901 AB#33546 - Filters Rail and Other Misc. Component Updates

### PR DESCRIPTION
**Pull request by morethanfire**
Took mostly what was done for Filters Rail in version 10.0.0 and we're putting it in version 9 in order to fix a few problems that have existed for quite some time.

There is some regression risk with this update. 

Components Updated:
- Filters Rail
- Container
- Content
- Page Action Bar
- Page Container
- Page Content
- Rail

Components Added:
- Slide

Others Updated:
- Theme Gutters

Docs Updated:
- Page Demo

### CODE REVIEW

**Comment by morethanfire** [comment]
Also this PR fixes #227

**Comment by morethanfire** [comment]
Please do not merge this as of right now, because it needs to be tested in HC and also refactor all areas using the root Container component to use the Page component.

File: docs/src/layout/page/demo/demo.jsx
Comment (groberts314) [inline docs/src/layout/page/demo/demo.jsx:1138]: Is this commented out for a reason?

File: src/surfaces/filtersRail/__test__/filtersRail.test.js
Comment (groberts314) [inline src/surfaces/filtersRail/__test__/filtersRail.test.js:47]: Curious: why this with no `expect()` or anything afterwards?

File: src/surfaces/filtersRail/__test__/filtersRail.test.js
Comment (groberts314) [inline src/surfaces/filtersRail/__test__/filtersRail.test.js:62]: Not a huge deal, but we might consider:
```JavaScript
const foo = wrapper.find('.foo');
expect(foo.exists()).toEqual(true);
expect(foo.text()).toEqual(text);
```

File: src/surfaces/filtersRail/__test__/filtersRail.test.js
Comment (groberts314) [inline src/surfaces/filtersRail/__test__/filtersRail.test.js:112]: Looks like Jest has a way of marking `todo` tests: https://jestjs.io/docs/en/api#testtodoname
_Not a Blocker_

File: src/surfaces/rail/rail.jsx
Comment (groberts314) [inline src/surfaces/rail/rail.jsx:63]: Is there a reason we're _not_ using a functional component here?

**Review by groberts314 — CHANGES_REQUESTED** [review]
Overall looks pretty good.  A few small observations.  We can negotiate NABs versus blockers and get to approval.

File: src/surfaces/rail/rail.jsx
Comment (morethanfire) [reply src/surfaces/rail/rail.jsx:63]: The only reason is because it's an old component. Do you think we should take the time - probably not much, that i know - to convert?

File: src/surfaces/filtersRail/__test__/filtersRail.test.js
Comment (morethanfire) [reply src/surfaces/filtersRail/__test__/filtersRail.test.js:112]: Hey, that's cool! Thanks.

File: docs/src/layout/page/demo/demo.jsx
Comment (morethanfire) [reply docs/src/layout/page/demo/demo.jsx:1138]: Yeah, I commented it out because, we need to remove it envtually by putting it maybe in a drawer or something. I don't know. I was just trying to make the demo page look sort of nice. It's not great.

File: src/surfaces/rail/rail.jsx
Comment (groberts314) [reply src/surfaces/rail/rail.jsx:63]: I think we could/should.  But if you'd rather hold off for the sake of expediency, I can understand that decision as well.

File: docs/src/layout/page/demo/demo.jsx
Comment (groberts314) [reply docs/src/layout/page/demo/demo.jsx:1138]: Got it.  Can we maybe add a comment?
```JavaScript
{/* TODO: Move this to a Drawer or something.  For now, commenting it out to clean up Demo Page */}
```

File: src/surfaces/rail/rail.jsx
Comment (morethanfire) [reply src/surfaces/rail/rail.jsx]: Fixed.

File: src/surfaces/filtersRail/__test__/filtersRail.test.js
Comment (morethanfire) [reply src/surfaces/filtersRail/__test__/filtersRail.test.js]: Changed.

File: src/surfaces/filtersRail/__test__/filtersRail.test.js
Comment (morethanfire) [reply src/surfaces/filtersRail/__test__/filtersRail.test.js]: Fixed.

File: src/surfaces/filtersRail/__test__/filtersRail.test.js
Comment (morethanfire) [reply src/surfaces/filtersRail/__test__/filtersRail.test.js:49]: Fixed.

File: docs/src/layout/page/demo/demo.jsx
Comment (morethanfire) [reply docs/src/layout/page/demo/demo.jsx:1142]: Added.

Comment (groberts314) [reply docs/src/layout/page/demo/demo.jsx:1142]: Gracias señor!  Resolving.

File: src/surfaces/filtersRail/__test__/filtersRail.test.js
Comment (groberts314) [reply src/surfaces/filtersRail/__test__/filtersRail.test.js:49]: Very nice!  Resolving.

File: src/surfaces/filtersRail/__test__/filtersRail.test.js
Comment (groberts314) [reply src/surfaces/filtersRail/__test__/filtersRail.test.js]: Me likey.  Resolving.

File: src/surfaces/filtersRail/__test__/filtersRail.test.js
Comment (groberts314) [reply src/surfaces/filtersRail/__test__/filtersRail.test.js]: Nice!  Resolving.

File: src/surfaces/rail/rail.jsx
Comment (groberts314) [reply src/surfaces/rail/rail.jsx]: Beauty.  Resolving.

**Review by groberts314 — APPROVED** [review]
Change requests all addressed.
:shipit: as soon as validation builds complete.

**Comment by groberts314** [comment]
Ugh, test failures: https://dev.azure.com/saddlebackchurch/Church%20Management/_build/results?buildId=23035&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b&t=ec339c2b-c439-539d-f401-d5569bbc95f9&l=962

File: src/surfaces/rail/rail.jsx
Comment (groberts314) [inline src/surfaces/rail/rail.jsx:62]: shouldn't we do the named function thingie...?

**Pull request merged**
